lib/services/notification-api-service.js

```diff
@@ -70,14 +72,47 @@ function notificationApiServiceFactory(
         });
     };
 
-    notificationApiService.prototype.postBroadcastNotification = function(message) {
-        var self = this;
-
+    NotificationApiService.prototype.postBroadcastNotification = function(message) {
         return eventsProtocol.publishBroadcastNotification(message)
         .then(function () {
             return message;
         });
     };
 
-    return new notificationApiService();
+    NotificationApiService.prototype.postProgressEvent = function(data) {
+        var progressData;
+        return waterline.taskdependencies.find({taskId: data.taskId})
+        .then(function(task){
+            if (!_.isEmpty(task) && _.has(task[0], 'graphId')){
+                return waterline.graphobjects.find({instanceId: task[0].graphId})
+                .then(function(graphObject){
+                    //TODO: workflow progress percentage should be designed
+                    progressData = {
+                        graphId: graphObject[0].instanceId,
+                        graphName: graphObject[0].definition.friendlyName,
+                        progress: {
+                            percentage: "na",
+                            description: data.progress.description
+                        },
+                        taskProgress: {
+                            graphId: graphObject[0].instanceId, 
+                            taskId: data.taskId,
+                            taskName: graphObject[0].tasks[data.taskId].friendlyName,
+                            progress: data.progress 
+                        }
+                    };
```

Comment:
suggest to return something in this .then block. One option would be combine following then block with this one.